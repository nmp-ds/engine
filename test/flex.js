import { expect, test } from 'vitest';
import { setup, getNumericArrayInRange } from './_helpers.js';

setup();

test('flex with arbitrary values', async (t) => {
  const classes = ['flex-[2_2_0%]'];

  const { css } = await t.uno.generate(classes);
  expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .flex-\\\\[2_2_0\\\\%\\\\]{flex:2 2 0%;}"
    `);
});

test('flex directions', async (t) => {
  const classes = ['flex-row', 'flex-row-reverse', 'flex-col', 'flex-col-reverse'];

  const { css } = await t.uno.generate(classes);
  expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .flex-row{flex-direction:row;}
      .flex-row-reverse{flex-direction:row-reverse;}
      .flex-col{flex-direction:column;}
      .flex-col-reverse{flex-direction:column-reverse;}"
    `);
});

test('flex wraps', async (t) => {
  const classes = ['flex-wrap', 'flex-wrap-reverse', 'flex-nowrap'];

  const { css } = await t.uno.generate(classes);
  expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .flex-wrap{flex-wrap:wrap;}
      .flex-wrap-reverse{flex-wrap:wrap-reverse;}
      .flex-nowrap{flex-wrap:nowrap;}"
    `);
});

test('flex vitals', async (t) => {
  const classes = ['flex-1', 'flex-auto', 'flex-initial', 'flex-none'];

  const { css } = await t.uno.generate(classes);
  expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .flex-1{flex:1 1 0%;}
      .flex-auto{flex:1 1 auto;}
      .flex-initial{flex:0 1 auto;}
      .flex-none{flex:none;}"
    `);
});

test('flex invalid', async (t) => {
  const classes = ['flex-0', 'flex-shrink', 'flex-shrink-0', 'flex-grow-0'];

  const { css } = await t.uno.generate(classes);
  expect(css).toMatchInlineSnapshot('""');
});

test('flex shrink/grow', async (t) => {
  const boundsRangeArray  = getNumericArrayInRange(0, 5);
  const autoClasses = ['grow', 'shrink'];
  const classes = boundsRangeArray.map((num) => [`shrink-${num}`, `grow-${num}`]).flat();

  const { css } = await t.uno.generate([...classes, ...autoClasses]);
  expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .shrink,
      .shrink-1{flex-shrink:1;}
      .shrink-0{flex-shrink:0;}
      .shrink-2{flex-shrink:2;}
      .shrink-3{flex-shrink:3;}
      .shrink-4{flex-shrink:4;}
      .shrink-5{flex-shrink:5;}
      .grow,
      .grow-1{flex-grow:1;}
      .grow-0{flex-grow:0;}
      .grow-2{flex-grow:2;}
      .grow-3{flex-grow:3;}
      .grow-4{flex-grow:4;}
      .grow-5{flex-grow:5;}"
    `);

});
