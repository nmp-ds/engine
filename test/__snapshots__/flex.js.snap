// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flex shrink/grow with arbitrary values 1`] = `
"/* layer: default */
.shrink,
.shrink-\\[1\\]{flex-shrink:1;}
.shrink-\\[0\\]{flex-shrink:0;}
.shrink-\\[2\\]{flex-shrink:2;}
.shrink-\\[3\\]{flex-shrink:3;}
.shrink-\\[4\\]{flex-shrink:4;}
.shrink-\\[5\\]{flex-shrink:5;}
.grow,
.grow-\\[1\\]{flex-grow:1;}
.grow-\\[0\\]{flex-grow:0;}
.grow-\\[2\\]{flex-grow:2;}
.grow-\\[3\\]{flex-grow:3;}
.grow-\\[4\\]{flex-grow:4;}
.grow-\\[5\\]{flex-grow:5;}"
`;

exports[`flex-basis 1`] = `
"/* layer: default */
.basis-1{flex-basis:0.1rem;}
.basis-1\\/2{flex-basis:50%;}
.basis-1\\/4{flex-basis:25%;}
.basis-1\\/5{flex-basis:20%;}
.basis-16{flex-basis:1.6rem;}
.basis-2{flex-basis:0.2rem;}
.basis-2\\/5{flex-basis:40%;}
.basis-32{flex-basis:3.2rem;}
.basis-4{flex-basis:0.4rem;}
.basis-48{flex-basis:4.8rem;}
.basis-64{flex-basis:6.4rem;}
.basis-8{flex-basis:0.8rem;}"
`;

exports[`flex-basis with arbitrary values 1`] = `
"/* layer: default */
.basis-\\[15\\]{flex-basis:1.5rem;}
.basis-\\[15\\%\\]{flex-basis:15%;}
.basis-\\[15px\\]{flex-basis:15px;}
.basis-\\[15rem\\]{flex-basis:15rem;}"
`;
